client/components/inline-support-link/index.jsx

```diff
@@ -0,0 +1,76 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import { openSupportArticleDialog } from 'state/inline-support-article/actions';
+
+class InlineSupportLink extends Component {
+	static propTypes = {
+		supportPostId: PropTypes.number.isRequired,
+		supportLink: PropTypes.string,
+		showText: PropTypes.bool,
+		text: PropTypes.string,
+		showIcon: PropTypes.bool,
+		iconSize: PropTypes.number,
+	};
+
+	static defaultProps = {
+		supportPostId: null,
+		supportLink: null,
+		showText: true,
+		text: null,
+		showIcon: true,
+		iconSize: 14,
+	};
+
+	handleClick = event => {
+		const { supportPostId, supportLink } = this.props;
+
+		event.preventDefault();
+		this.props.openSupportArticleDialog( { postId: supportPostId, postUrl: supportLink } );
+	};
+
+	render() {
+		const {
+			showText,
+			text,
+			supportPostId,
+			supportLink,
+			showIcon,
+			iconSize,
+			translate,
+		} = this.props;
+
+		if ( ! supportPostId ) {
```

Comment:
A thought I had here is that we could fallback to just an external link in cases where the `supportPostId` is missing for some reason but the `supportLink` is present - something that might happen in cases where the implementation is dynamic. Happy to explore that in a later iteration though :)

Reply:
I actually had this in an earlier version - I think it's a fair idea, and shouldn't be hard to add back in.

Reply:
Added back in db61239